core/src/banking_stage.rs

```diff
@@ -376,6 +377,11 @@ impl BankingStage {
                     batch_limit,
                     transaction_status_sender.clone(),
                 );
+                if decision == BufferedPacketsDecision::Hold {
```

Comment:
@carllin I simplified the logic here. I think in the case we are holding, we can just fall out of the loop to ping the transaction receiver. If we are forwarding we should clear the buffered list after we send it to the next leader. If we are consume, we just stay in the loop to keep clearing the buffered packets.

Reply:
oh I was wondering where all the code went 😄